### PR TITLE
python-package-available

### PR DESCRIPTION
**Pull request by thautwarm**
Some examples:
```python

import GraphEngine as ge
gm = ge.GraphMachine('storage')
gm.start()

from GraphEngine.Storage.core.Cell import Cell
from GraphEngine.Storage.core.SymTable import load_symbols, sync
from GraphEngine.Storage.core.SymTable import SymTable

load_symbols('tests/tsl/')  # dynamically load TSL
print(SymTable.inst)  # cell type symbol table


# take the Type `C1` from symtable, and use it to new a cell.
c1 = Cell(SymTable.inst['C1'])

c1.set('foo', 1)
print(c1)

c1.compute()

c1.set('foo', 2)
print(c1)
```
Here is something very exciting: although we're writing GE in python, we're indeed using type checking! It means that we could not use `c1.set('foo', '2')` , only `c1.set('foo', 2)` is allowed! The type system could be scalable, and it even supports generic types now.

However, we need more test examples now...

### CODE REVIEW

File: src/Modules/Trinity.FFI/Trinity.FFI/ObjectStore.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        private const int c_default_len = 1 << 20;
+
+        public ObjectStore()
+        {
+            m_lock = new object();
+            _Alloc_Init(c_default_len);
+            
+        }
+        public Action<int> _Alloc_Mutable;
```
Comment (yatli) [inline src/Modules/Trinity.FFI/Trinity.FFI/ObjectStore.cs:47]: why extra function pointer?

File: src/Modules/Trinity.FFI/Trinity.FFI/ObjectStore.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            m_array = ret;
+            m_len = m_array.Length;
+        }
+
+        private void _Alloc(int newlen)
+        {
+            _Alloc_Mutable(newlen);
+        }
+
+        public void DeAlloc()
```
Comment (yatli) [inline src/Modules/Trinity.FFI/Trinity.FFI/ObjectStore.cs:84]: why public DeAlloc? It means that a user could deallocate the references without disposing them.

File: src/Modules/Trinity.FFI/Trinity.FFI/TrinityWrapper.cs [hunk 3]
```diff
@@ -159,6 +160,36 @@ public static TrinityErrorCode trinity_ffi_local_savecell_2(long cellId, CellAcc
                 return TrinityErrorCode.E_FAILURE;
             }
         }
+
+        public static TrinityErrorCode trinity_ffi_local_savecell_3(IntPtr cell)
+        {
+            try
+            {
+                ICell c = (ICell)GCHandle.FromIntPtr(cell).Target;
```
Comment (yatli) [inline src/Modules/Trinity.FFI/Trinity.FFI/TrinityWrapper.cs:168]: might help to further improve performance if we do ObjectStore right from here :) Remember, our goal is to minimize cross-runtime calls and GCHandle. So it does not make a lot of sense to go back and forth between a foreign runtime and clr, to put an object into the cache

**Review by yatli — CHANGES_REQUESTED** [review]
hey red that's some really nice-looking example up there! good job! I've added a few comments and we can discuss about them together.

**Comment by yatli** [comment]
CI servers down?

**Comment by yatli** [comment]
@thautwarm why implement IDAllocator in python?

File: src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellManager.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            private int moduleId;
+            public int ModuleId { get => moduleId;}
+
+            static ObjectStore<ICell> Cache = new ObjectStore<ICell> { };
+
+            
+
+            public CellManager()
+            {
+                Cache = new ObjectStore<ICell> { };
```
Comment (yatli) [inline src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellManager.cs:26]: do not allocate static member in a ctor.

File: src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellAccessorManager.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
```
Comment (yatli) [inline src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellAccessorManager.cs:1]: I'm still wondering why do we favor pythonnet over our custom method thunks

**Review by yatli — CHANGES_REQUESTED** [review]

File: src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellManager.cs (hunk 4 above)
Comment (thautwarm) [reply src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellManager.cs:26]: @yatli `IDAllocator` is to assign an ID for a cell... We talked about it that day and the automatically id assignment of Trinity's IDFactory didn't work until now.  
I thought you might have better ideas to solve automatically ID assignment, so I didn't make additional changes in .NET...

File: src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellAccessorManager.cs (hunk 5 above)
Comment (thautwarm) [reply src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellAccessorManager.cs:1]: This is just the slow version for Python binding.  
I'm to implement a fast one with gasm, and then a new version totally written with c-extensions would be available.  
During the lasy two days I'm implementing gasm with cpp...

Comment (thautwarm) [reply src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellAccessorManager.cs:1]: The slow version is mainly for data transformation.  
When you have to load data from Graph Engine servers, just use current implementation and you will get original python obejcts.

File: src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellManager.cs (hunk 4 above)
Comment (yatli) [reply src/Modules/Trinity.FFI/Trinity.FFI/Agent/Storage.CellManager.cs:26]: @thautwarm got you. I'll work on distributed consistent id allocator.

**Review by yatli — APPROVED** [review]

**Pull request merged**